controllers/cassandrabackup_controller.go

```diff
@@ -69,17 +71,62 @@ func (r *CassandraBackupReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 
 	backup := instance.DeepCopy()
 
+	// Verify the CassandraBackup has a finalizer
+	if !ctrlutil.ContainsFinalizer(backup, finalizerName) {
+		ctrlutil.AddFinalizer(backup, finalizerName)
+		err = r.Update(ctx, backup)
+		if err != nil {
+			return ctrl.Result{RequeueAfter: 10 * time.Second}, err
+		}
+	}
+
+	// If there is anything in progress, simply requeue the request
+	if len(backup.Status.InProgress) > 0 || backup.Status.DeletionInProgress {
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	if backup.GetDeletionTimestamp() != nil {
+		if ctrlutil.ContainsFinalizer(backup, finalizerName) {
+			// Deletion process has already been started, return later
+			patch := client.MergeFrom(backup.DeepCopy())
+			backup.Status.DeletionInProgress = true
+
+			if err := r.Status().Patch(context.Background(), backup, patch); err != nil {
+				r.Log.Error(err, "failed to patch status with backup deletion inprogress status", "Backup", backup)
+				return ctrl.Result{Requeue: true, RequeueAfter: 5 * time.Second}, err
+			}
+
+			go func() {
+				// Backup is being deleted, call Medusa to remove it also from storage
+				err = r.removeBackup(ctx, backup)
+				if err != nil {
+					r.Log.Error(err, "failed to remove backup", "Backup", backup)
```

Comment:
If `removeBackup` fails then as a user I might find it confusing to see that `deletionInProgress` has been reset and that the backup still exists. Do you think we need something else in the status to communicate the state?